src/core/js/tracking.js

```diff
@@ -0,0 +1,123 @@
+define([
+    'core/js/adapt',
+    'core/js/enums/completionStateEnum'
+], function(Adapt, COMPLETION_STATE) {
+
+    var Tracking = Backbone.Controller.extend({
+
+        _config: {
+            _requireContentCompleted: true,
+            _requireAssessmentPassed: false
+        },
+
+        _completionCriteria: [],
+
+        initialize: function() {
+            Adapt.once('configModel:loadCourseData', this.loadConfig.bind(this));
+            Adapt.on('app:dataReady', this.setupEventListeners.bind(this));
+        },
+
+        setupEventListeners: function() {
+            var self = this;
```

Comment:
self isn't needed, the ``this.listenTo`` function returns with the correct this context, that is it's function